spec/requests/event_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Event' do
+  describe 'POST /events' do
+    let(:event_params) do
+      {
+        event: {
+          title: 'Morning meeting',
+          description: 'Meeting on current project',
+          date: Date.tomorrow
+        }
+      }
+    end
+    let(:invalid_params) do
+      {
+        event: {
+          title: 'Morning meeting',
+          description: 'Meeting on current project'
+        }
+      }
+    end
+
+    it 'creates event record' do
+      expect do
+        post '/events', params: event_params
+      end.to change(Event, :count).by(1)
+    end
+
+    it 'returns "created" status' do
+      post '/events', params: event_params
+
+      expect(response).to have_http_status(:created)
+    end
+
+    it 'returns informative message' do
+      post '/events', params: event_params
+
+      expect(json).to eq('Event was successfully created')
+    end
+
+    it 'Do not create event without date' do
+      expect do
+        post '/events', params: invalid_params
+      end.not_to change(Event, :count)
+    end
+
+    it 'returns "unprocessable_entity" status' do
+      post '/events', params: invalid_params
+
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+  end
+
+  describe 'GET /enents/:event_id' do
+    let!(:event) { create(:event) }
+
+    it 'returns json representation of the event' do
+      get "/events/#{event.id}"
```

Comment:
You can put this in before block instead of calling in all tests

Reply:
If I use before block:
```
before do
      @event = create(:event)
    end

    it 'returns json representation of the event' do
      get "/events/#{@event.id}"
...
```
the rubocop don't pass and has offenses:
```
spec/requests/event_spec.rb:67:22: C: RSpec/InstanceVariable: Avoid instance variables – use let, a method call, or a local variable (if possible).
      get "/events/#{@event.id}"
                             ^^^^^^
```
So leave it in let block.